scalalib/src/JavaModule.scala

```diff
@@ -563,12 +576,29 @@ trait JavaModule
         coursierCacheCustomizer = coursierCacheCustomizer()
       )
 
+      val roots = whatDependsOn match {
+        case List() => flattened
+        case _ =>
+          // We don't really care what scalaVersions is set as here since the user
+          // will be passing in `_2.13` or `._3` anyways. Or it may even be a java
+          // dependency. Looking at the usage upstream, it seems that this is set if
+          // it can be or else defaults to "". Using it, I haven't been able to see
+          // any difference whether or not it's set, and by using "" it greatly simplifies
+          // it.
+          val matchers = whatDependsOn
+            .map(module => module.module(scalaVersion = ""))
+            .map(module => ModuleMatcher(module))
+
+          resolution.minDependencies
+            .filter(dep => matchers.exists(matcher => matcher.matches(dep.module))).toSeq
+      }
+
       println(
         coursier.util.Print.dependencyTree(
-          roots = flattened,
           resolution = resolution,
+          roots = roots,
           printExclusions = false,
-          reverse = inverse
+          reverse = if (whatDependsOn.isEmpty) inverse else true
```

Comment:
I was literally stuck on this for hours today. It totally didn't click why this wasn't working for me, and then I realized in order for the entire thing to work, it's just a trick. We target the deps that we are looking for as the roots, and then we inverse it. I wasn't doing that and couldn't figure out why it wasn't working. I made a not above, but if `whatDependsOn` exists, this _has to be true_.

Reply:
We should add to the (scala)doc, that `whatDependsOn` always forces `inverse` enabled.

Reply:
I had a note about it in the `@param whatDependsOn`, but I've moved it up into the general doc for better visibility.